src/TestUtils/src/UITest.Appium/AppiumAndroidApp.cs

```diff
@@ -44,26 +44,33 @@ public override ApplicationState AppState
 		{
 			get
 			{
-				var appId = Config.GetProperty<string>("AppId") ?? throw new InvalidOperationException($"{nameof(AppState)} could not get the appid property");
-				var state = _driver?.ExecuteScript("mobile: queryAppState", new Dictionary<string, object>
+				try
+				{
+					var appId = Config.GetProperty<string>("AppId") ?? throw new InvalidOperationException($"{nameof(AppState)} could not get the appid property");
+					var state = _driver?.ExecuteScript("mobile: queryAppState", new Dictionary<string, object>
 						{
 							{ "appId", appId },
 						});
 
-				// https://github.com/appium/appium-uiautomator2-driver#mobile-queryappstate
-				if (state == null)
+					// https://github.com/appium/appium-uiautomator2-driver#mobile-queryappstate
+					if (state == null)
+					{
+						return ApplicationState.Unknown;
+					}
+
+					return Convert.ToInt32(state) switch
+					{
+						0 => ApplicationState.Not_Installed,
+						1 => ApplicationState.Not_Running,
+						3 or
+						4 => ApplicationState.Running,
+						_ => ApplicationState.Unknown,
+					};
+				}
+				catch
```

Comment:
can we log the exception?

Reply:
Why do we need to add this try catch? I think the AppSetup does handle this logic of retry and log